src/hivpy/demographics.py

```diff
@@ -246,3 +250,21 @@ def determine_deaths(self, population_data: pd.DataFrame) -> pd.Series:
                 # Mark those who died from this group
                 all_died |= pd.Series(died, index=group.index)
         return population_data.date_of_death.isnull() & all_died
+
+    def determine_deaths(self, population_data: pd.DataFrame) -> pd.Series:
+        """Get which individuals die in a time step, as a boolean Series."""
+        # This binning should perhaps happen when the date advances
+        # Age groups are the same regardless of sex
+        age_limits = [lower for (lower, upper) in DEATH_RATE_MALE] + [inf]
+        population_data["age_group"] = np.digitize(population_data.age, age_limits)
+
+        death_probs = np.full(len(population_data), np.nan)
+        groups = population_data.groupby(["sex", "age_group"])
+        for (sex, age_group), entries in groups.groups.items():
```

Comment:
maybe rename `groups` so we don't have the slightly confusing `groups.groups.items()`